misc-tools/import-accounts-multiuser.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+import sys, csv, os, re, json
+
+api_url = sys.argv[1]
+httpie_options = sys.argv[2]
+
+# Note: this script uses httpie so we can use the same commandline flags as the base import-contest.sh script
+
+# Import all non-team accounts. We use the existing CSV import functionality for this,
+# because for jury members it creates teams and a category, which we can only do in the
+# API if we know a contest ID. We do not know one at this point
+def import_non_team_accounts():
+    with open('accounts.tsv') as csv_file:
+        with open('accounts_non_team.tsv', 'w') as csv_output:
+            csv_reader = csv.reader(csv_file, delimiter="\t")
+            csv_writer = csv.writer(csv_output, delimiter="\t")
+            first_line = True
+            for row in csv_reader:
+                # Ignore first line of the CSV
+                if first_line:
+                    csv_writer.writerow(row)
+                    first_line = False
+                    continue
+                # We are not interested in analysts currently
+                if row[0] == "team":
+                    continue
+                csv_writer.writerow(row)
+
+    command = "http {} --check-status -b -f POST \"{}/users/accounts\" tsv@accounts_non_team.tsv".format(httpie_options, api_url)
+    os.system(command)
+
+# Get all team accounts
+def get_team_accounts():
+    accounts = []
+    with open('accounts.tsv') as csv_file:
+        csv_reader = csv.reader(csv_file, delimiter="\t")
+        first_line = True
+        for row in csv_reader:
+            # Ignore first line of the CSV
+            if first_line:
+                first_line = False
+                continue
+            # We are not interested in analysts currently
+            if row[0] != "team":
+                continue
+            accounts.append({
+                "role": row[0],
+                "name": row[1],
+                "username": row[2],
+                "password": row[3]
+            })
+
+    return accounts
+
+import_non_team_accounts()
+
+team_regex = r"^[^0-9]*0*([0-9]+)([^0-9]*)$"
+accounts = get_team_accounts()
+for account in accounts:
+    team_data = re.search(team_regex, account["username"])
+    team_id = team_data.group(1)
+    user_suffix = team_data.group(2)
+    ip_mapping = {
```

Comment:
Where did we take the IP addresses from before this script?

Reply:
```sql
UPDATE user set IP = concat('10.1.1.', team_id) WHERE team_id IS NOT NULL
```

I think we did something ugly like this.

Reply:
Can we add an optional column to accounts.tsv instead with the IP address mapping?

Reply:
I can add it. What if it is not present, set no IP?